src/source/source_cache.test.ts

```diff
@@ -1184,8 +1185,40 @@ describe('SourceCache#_updateRetainedTiles', () => {
 
     });
 
-    test('Only retain loaded parent tile when zooming in', () => {
+    test('Retain, then cancel loading tiles when zooming in', () => {
         const sourceCache = createSourceCache();
+        // Disabling pending tile canceling (thus retaining) in Map mock:
+        const map = {cancelPendingTileRequestsWhileZooming: false} as Map;
+        sourceCache.onAdd(map);
+        sourceCache._source.loadTile = async (tile) => {
+            tile.state = 'loading';
+        };
+
+        let idealTiles = [new OverscaledTileID(9, 0, 9, 0, 0), new OverscaledTileID(9, 0, 9, 1, 0)];
+        sourceCache._updateRetainedTiles(idealTiles, 9);
+        idealTiles = [new OverscaledTileID(10, 0, 10, 0, 0), new OverscaledTileID(10, 0, 10, 1, 0)];
+        let retained = sourceCache._updateRetainedTiles(idealTiles, 10);
+        expect(Object.keys(retained).sort()).toEqual([
+            new OverscaledTileID(9, 0, 9, 0, 0).key,    // retained
+            new OverscaledTileID(10, 0, 10, 0, 0).key,
+            new OverscaledTileID(10, 0, 10, 1, 0).key
+        ]);
+
+        // Canceling pending tiles now via runtime map property:
+        map.cancelPendingTileRequestsWhileZooming = true;
+        retained = sourceCache._updateRetainedTiles(idealTiles, 10);
+        // Parent loading tiles from z=9 not retained:
+        expect(Object.keys(retained).sort()).toEqual([
+            new OverscaledTileID(10, 0, 10, 0, 0).key,
+            new OverscaledTileID(10, 0, 10, 1, 0).key
+        ]);
+    });
+
+    test('Cancel or retain loading tiles when zooming in depending on current configuration', () => {
```

Comment:
```suggestion
    test('Cancel, then retain loading tiles when zooming in', () => {
```
So that it will be aligned with the previous test...?

Reply:
Well it'd be more like "Cancel, then retain, then cancel again..."  :) Maybe too much

Reply:
Updated my suggestion - this only tells us that the tests is testing "too much" :-)